pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -173,6 +169,113 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 	return resources, model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", cacheStats.hits, cacheStats.hits+cacheStats.miss)}
 }
 
+func (configgen* ConfigGeneratorImpl) generateDeltaServices(updatedService model.ConfigKey, proxy *model.Proxy, push *model.PushContext, watched []string) ([]*model.Service, []string) {
+	deletedClusters := make([]string, 0)
+	services := make([]*model.Service, 0)
+	service := push.ServicesForHostname(proxy, host.Name(updatedService.Name))
+	// SidecarScope.ServiceForHostname will return nil if the proxy doesn't care about the service OR it was deleted.
+	// we can cross-reference with WatchedResources to figure out which services were deleted.
+	if service == nil {
+		// we assume a service was deleted, it doesn't hurt to have positives for removedResources when a resource included
+		// does not exist.
+		// WatchedResources.ResourceNames will contain the names of the clusters it is subscribed to. We can
+		// check with the name of our service (cluster names are in the format outbound|<port>||<hostname>
+		// so, we can check if the cluster names contains the service name, and determine if it is deleted by that
+		deletedClusters = append(deletedClusters, getRemovedClusterNames(watched, nil, updatedService.Name)...)
+	} else {
+		services = append(services, service...)
+	}
+	return services, deletedClusters
+}
+
+func (configgen *ConfigGeneratorImpl) generateDeltaDestinationRules(updatedDr model.ConfigKey, proxy *model.Proxy, push *model.PushContext, watched []string) ([]*model.Service, []string) {
```

Comment:
nit: we are not generating delta destinationrules may be some thing like deltaServicesFromDestinationRules?